### PR TITLE
Fixing the issue that FLAML trial number is significantly smaller than Transformers.hyperparameter_search

### PR DESCRIPTION
**Pull request by liususan091219**
1. Pad to multiples of 8 instead of the max_seq_length to avoid the long runtime for loss.backward()
2. Downgrade Transformers version to 4.18 to avoid the Transformers bug of not updating batch size in each trial. 
3. Adding low cost init config for TransformersEstimator

### CODE REVIEW

File: flaml/nlp/huggingface/utils.py [hunk 1]
```diff
@@ -21,15 +28,20 @@ def tokenize_text(X, Y=None, task=None, hf_args=None, tokenizer=None):
...
         )
     elif task in NLG_TASKS:
         return tokenize_seq2seq(X, Y, tokenizer=tokenizer, task=task, hf_args=hf_args)
     elif task == MULTICHOICECLASSIFICATION:
-        return tokenize_text_multiplechoice(X, tokenizer=tokenizer, hf_args=hf_args)
+        X_tokenized = tokenize_text_multiplechoice(
+            X, tokenizer=tokenizer, hf_args=hf_args
+        )
+        Y = todf(X_tokenized, Y, ["label"])
+        return X_tokenized, Y
```
[the remark this answers is not in the document]
Comment (liususan091219) [reply flaml/nlp/huggingface/utils.py:49]: added to the docstr of todf()

File: flaml/ml.py [hunk 2]
```diff
@@ -152,86 +151,61 @@ def metric_loss_score(
...
+            metric_mode = huggingface_metric_to_mode[datasets_metric_name]
+
+            if metric_name.startswith("seqeval"):
+                y_processed_true = [
+                    [labels[tr] for tr in each_list] for each_list in y_processed_true
+                ]
+            elif metric in ("pearsonr", "spearmanr"):
+                y_processed_true = (
+                    y_processed_true.to_list()
+                    if type(y_processed_true) == pd.Series
```
Comment (sonichi) [inline flaml/ml.py:170]:
```suggestion
                    if isinstance(y_processed_true, pd.Series)
```

File: flaml/model.py [hunk 3]
```diff
@@ -526,18 +524,16 @@ def _model_init(self):
...
 
-        if (self._task not in NLG_TASKS) and (self._task != TOKENCLASSIFICATION):
-            processed_X, _ = self._preprocess(X=X, **self._kwargs)
-            processed_y = y
-        else:
-            processed_X, processed_y = self._preprocess(X=X, y=y, **self._kwargs)
+        processed_X, processed_y_df = self._tokenize_text(X=X, y=y, **self._kwargs)
+        processed_y = processed_y_df.iloc[
+            :, 0
+        ]  # convert y from pd.DataFrame back to pd.Series
```
Comment (sonichi) [inline flaml/model.py:533]: move the location of this comment and put the statement in one line.

File: flaml/model.py [hunk 4]
```diff
@@ -574,14 +570,25 @@ def tokenizer(self):
...
-            if self._task in (MULTICHOICECLASSIFICATION, TOKENCLASSIFICATION)
-            else None
-        )
+        data_collator_class = task_to_datacollator_class.get(self._task)
+
+        if data_collator_class:
+            kwargs = {
+                "model": self._model_init(),  # need to set model, or there's ValueError: Expected input batch_size (..) to match target batch_size (..)
+                "label_pad_token_id": -100,
+                "pad_to_multiple_of": 8,
```
Comment (sonichi) [inline flaml/model.py:579]: If there is a non-trivial reason of using these values, write comments. I can't tell whether they are right.

**Review by sonichi — APPROVED** [review]

**Pull request merged**